src/stackwalk.c

```diff
@@ -1108,6 +1108,42 @@ JL_DLLEXPORT void jl_print_backtrace(void) JL_NOTSAFEPOINT
     jlbacktrace();
 }
 
+JL_DLLEXPORT void jl_print_task_backtraces(void)
+{
+    jl_task_t *t;
+    jl_ptls_t ptls;
+    arraylist_t *live_tasks;
+    size_t i, j, n;
+
+    for (i = 0; i < jl_n_threads; i++) {
+        ptls = jl_all_tls_states[i];
+        live_tasks = &ptls->heap.live_tasks;
+        n = live_tasks->len;
+        jl_safe_printf("==== Thread %d (%zu live tasks)\n",
+                ptls->tid + 1, n + 1);
+        t = ptls->root_task;
+        jl_safe_printf("     ---- Root task (%p)\n", t);
+        jl_safe_printf("          (sticky: %d, started: %d, state: %d, tid: %d)\n",
+                t->sticky, t->started, jl_atomic_load_relaxed(&t->_state),
+                jl_atomic_load_relaxed(&t->tid) + 1);
+        jlbacktracet(t);
```

Comment:
Will it ever be possible for the root task's sttkbuf to be NULL? I think probably not, but is it worth guarding against it, here?

Reply:
`jlbacktracet(t)` calls `jl_rec_backtrace()` which [checks](https://github.com/JuliaLang/julia/blob/master/src/stackwalk.c#L866-L867).